pkg/controller/deployment/generic_deployment_controller.go

```diff
@@ -48,5 +50,6 @@ func newGenericDeploymentController(
 		withUnsupportedArgsOverrideHook,
 		withProxyEnv,
 		withCAConfigMap(kubeInformersForTargetNamespace.Core().V1().ConfigMaps(), deployment, trustedCAConfigmapName),
+		withCloudCredentials(kubeInformersForTargetNamespace.Core().V1().Secrets(), configInformer.Config().V1().Infrastructures(), deployment.Name),
```

Comment:
Add the secrets informer to the informer array on line 39.

Not only it will add a wait for its caches to sync upon the controller startup, it's also going to add event handlers to the informer events. Without it, your controller is not going to react to the secret changes.